CMakeLists.txt

```
@@ -239,6 +239,17 @@ add_library(_lightgbm SHARED src/c_api.cpp src/lightgbm_R.cpp ${SOURCES})
 
 if(MSVC)
     set_target_properties(_lightgbm PROPERTIES OUTPUT_NAME "lib_lightgbm")
+    if(USE_DEBUG)
+      set_property(
+        TARGET _lightgbm
+        PROPERTY MSVC_RUNTIME_LIBRARY "MultiThreadedDebugDLL"
+      )
```

Maybe use `set_target_properties` for the consistency and as more specific function than general `set_property`?
```suggestion
      set_target_properties(_lightgbm PROPERTIES MSVC_RUNTIME_LIBRARY "MultiThreadedDebugDLL")
```